Source/towners.cpp

```diff
@@ -594,20 +594,15 @@ void __cdecl ProcessTowners()
 
 ItemStruct *__fastcall PlrHasItem(int pnum, int item, int *i)
 {
-	unsigned int v3; // eax
-	int v4;          // ecx
-
-	v3 = 21720 * pnum;
-	*i = 0;
 	if (plr[pnum]._pNumInv <= 0)
-		return 0;
-	while (*(int *)((char *)&plr[0].InvList[*i].IDidx + v3) != item) {
-		v4 = *i + 1;
-		*i = v4;
-		if (v4 >= plr[v3 / 0x54D8]._pNumInv)
-			return 0;
+		return NULL;

```

Comment:
Yeah I considered this but the loop was already a while loop and not do-while. Also there is a test and load of the 6 and 16. I guess this could be because our compiler some how is able to know that _pNumInv is never 0 or less, but it's and int so I'm not sure how it's able to make this conclusion.

Reply:
Never mind I figured it out...